### PR TITLE
Upgrade notes for Forms/Deploy 11

### PR DESCRIPTION
**Pull request by AndyButland**
This adds the documentation about Forms and Deploy 11, mostly just adding the breaking changes for reference.

Now the first release candidates are out, these can be reviewed and published at any time.

### CODE REVIEW

File: Add-ons/Umbraco-Deploy/Extending/index.md [hunk 1]
```diff
@@ -466,12 +466,14 @@ _transferEntityService.RegisterTransferEntityType(
...
 ```
 
-If access to any services is required when parsing the entity Id, the `HttpContext` is provided as a parameter to the function. A service can be accessed from this, e.g.:
+If access to any services is required when parsing the entity Id, where the `HttpContext` is provided as a parameter to the function, a service can be retrieved. For example:
 
 ```c#
 var localizationService = httpContext.RequestServices.GetRequiredService<ILocalizationService>();
 ```
 
+Note that the `HttpContext` parameter for the `matchesRoutePath` and `matchesNodeId` functions was added in Deploy 11. Before that version it is necessary to use the `StaticServiceProvider.Instance` to access registered services or the `HttpContext`.
```
Comment (eshanrnh) [inline Add-ons/Umbraco-Deploy/Extending/index.md:475]:
```suggestion
:::note
The `HttpContext` parameter for the `matchesRoutePath` and `matchesNodeId` functions was added in Deploy 11. Before that version, it is necessary to use the `StaticServiceProvider.Instance` to access registered services or the `HttpContext`.
:::
```

File: Add-ons/Umbraco-Deploy/Upgrades/version-specific.md [hunk 2]
```diff
@@ -9,6 +9,34 @@ meta.Description: "Version specific documentation for upgrading to new major ver
...
 
+## Version 11
+
+Version 11 of Umbraco Deploy has a minimum dependency on Umbraco CMS core of `11.0.0`. It runs on .NET 7.
+
+The forms deployment component has a minimum dependency on Umbraco Forms of `11.0.0`.
+
+### Breaking changes
+
+Version 11 contains a number of breaking changes but we won't expect many projects to be affected by them as they are in areas that are not typical extension points.  For reference though, the full details are listed here.
```
Comment (eshanrnh) [inline Add-ons/Umbraco-Deploy/Upgrades/version-specific.md:20]:
```suggestion
Version 11 contains a number of breaking changes. We don't expect many projects to be affected by them as they are in areas that are not typical extension points.  For reference though, the full details are listed here:
```

File: Add-ons/Umbraco-Deploy/Upgrades/version-specific.md [hunk 3]
```diff
@@ -9,6 +9,34 @@ meta.Description: "Version specific documentation for upgrading to new major ver
...
+
+#### Code
+
+- The property `PreValues` on `FormArtifact` was changed from an `IEnumerable<string>`  to an `IEnumerable<FieldPrevalue>`, where `FieldPrevalue` contains a `Value` and `Caption`.
+- Nullable checks were enabled in the `Umbraco.Deploy.Forms` project and issues resolved by applying appropriate nullable settings to various properties.
+- The `TreeNodeGetter` function set as a property on `DeployTransferRegisteredEntityTypeDetail.RemoteTreeDetail` now takes a non-nullable `HttpContext`
+parameter.
+- The `matchesRoutePath` and `matchesNodeId` parameters provided to `ITransferEntityService.RegisterTransferEntityType`, and which populate the `MatchesRoutePath` and `MatchesNodeId` properties on `DeployTransferRegisteredEntityTypeDetail`, now take an HttpContext argument.
+- The `MultiNodeTreePickerPreValueConnector` was removed (as the format for Umbraco 8+ is as UDIs, and hence there is no processing to do).
+- Obsolete constructors, properties and methods on `Manifest` were removed.
```
Comment (eshanrnh) [inline Add-ons/Umbraco-Deploy/Upgrades/version-specific.md:30]:
```suggestion
- Obsolete constructors, properties, and methods on `Manifest` were removed.
```

File: Add-ons/Umbraco-Deploy/Upgrades/version-specific.md [hunk 4]
```diff
@@ -9,6 +9,34 @@ meta.Description: "Version specific documentation for upgrading to new major ver
...
+- Namespace of the `Package` class was adjusted and the obsolete property `Artifacts` was removed.  `ArtifactsWithOptions` was renamed to `Artifacts`.
+- The signature of `EnvironmentController.BeginCreateManifestForUdis` was changed.
+- The temporary interface `IUmbracoEnvironmentWithOptionsAwareManifest` was removed and elements added to `IUmbracoEnvironment`.
+- The signature of `IWorkItemFactory.CreateSourceDeplo`y was changed to accommodate culture and scheduled publishing options.
+- The obsolete constructor on `SourceDeployWorkItem` was removed.
+- The class `UmbracoFormsCompatibilit`y that is no longer required has been removed.
```
Comment (eshanrnh) [inline Add-ons/Umbraco-Deploy/Upgrades/version-specific.md:36]:
```suggestion
- The class `UmbracoFormsCompatibility` that is no longer required has been removed.
```

File: Add-ons/Umbraco-Deploy/Upgrades/version-specific.md [hunk 5]
```diff
@@ -9,6 +9,34 @@ meta.Description: "Version specific documentation for upgrading to new major ver
...
+- Obsolete constructors were removed on `NoNodesController`, `UiController` and `UiControllerBase`.
```
Comment (eshanrnh) [inline Add-ons/Umbraco-Deploy/Upgrades/version-specific.md:37]:
```suggestion
- Obsolete constructors were removed on `NoNodesController`, `UiController`, and `UiControllerBase`.
```

File: Add-ons/UmbracoForms/Installation/Version-Specific.md [hunk 6]
```diff
@@ -7,6 +7,45 @@ versionTo: 10.0.0
 
 This page covers specific upgrade documentation for specific versions.
 
+## Version 11
+
+### 11.0
+
+Version 10 of Umbraco Forms has a minimum dependency on Umbraco CMS core of `11.0.0`. It runs on .NET 7.
```
Comment (eshanrnh) [inline Add-ons/UmbracoForms/Installation/Version-Specific.md:14]:
```suggestion
Version 11 of Umbraco Forms has a minimum dependency on Umbraco CMS core of `11.0.0`. It runs on .NET 7.
```

File: Add-ons/UmbracoForms/Installation/Version-Specific.md [hunk 7]
```diff
@@ -7,6 +7,45 @@ versionTo: 10.0.0
...
+
+#### Breaking changes
+
+Version 11 contains a number of breaking changes but we won't expect many projects to be affected by them as they are in areas that are not typical extension points. For reference though, the full details are listed here.
```
Comment (eshanrnh) [inline Add-ons/UmbracoForms/Installation/Version-Specific.md:18]:
```suggestion
Version 11 contains a number of breaking changes. We don't expect many projects to be affected by them as they are in areas that are not typical extension points. For reference though, the full details are listed here:
```

File: Add-ons/UmbracoForms/Installation/Version-Specific.md [hunk 8]
```diff
@@ -7,6 +7,45 @@ versionTo: 10.0.0
...
+
+##### Presentation
+
+- A CSS class for each field is rendered out matching the caption of the form field.  This has been changed to use the alias of the form field, as this is considered less likely to change and potentially break custom styles.
+
+##### Code
+
+- The int `DeleteFormRecords(Form form, FormState formState, DateTime deleteRecordsCreatedEarlierThan)` method was added to `IRecordStorage`.
+- Name of` FormsUmbracoBuiderExtensions` was corrected to `FormsUmbracoBuilderExtensions`.
```
Comment (eshanrnh) [inline Add-ons/UmbracoForms/Installation/Version-Specific.md:27]:
```suggestion
- Name of `FormsUmbracoBuiderExtensions` was corrected to `FormsUmbracoBuilderExtensions`.
```

File: Add-ons/UmbracoForms/Installation/Version-Specific.md [hunk 9]
```diff
@@ -7,6 +7,45 @@ versionTo: 10.0.0
...
+- The method `RegenerateFormStructureIds` on Form was amended to return a response.
+- The method void `ExecuteWorkflows(List<IWorkflow> workflows, Record record, Form form, FormState state)` was added to `IWorkflowExecutionService`.
+- Obsolete constructor on `PlaceholderParsingService` removed.
+- Obsolete constructor on `ServerVariablesParsingHandler` removed.
+- `IsMandatory` and `Condition` properties were added to the `IWorkflow` and `IWorkflowEntity` interface.
+- `DaysToRetainSubmittedRecordsFor` and `DaysToRetainApprovedRecordsForproperties` were added to the `IFormEntity` interface.
+- Obsolete constructor on the export type `ExportToExcel` removed.
+- Obsolete constructor on the workflow type `SendRazorEmail` removed.
+- Obsolete constructor on the controllers `UmbracoFormsController`, `ExportController`, `FieldController`, `FormController`, `RecordController` and  `EmailTemplateTreeController` removed.
```
Comment (eshanrnh) [inline Add-ons/UmbracoForms/Installation/Version-Specific.md:36]:
```suggestion
- Obsolete constructor on the controllers `UmbracoFormsController`, `ExportController`, `FieldController`, `FormController`, `RecordController`, and  `EmailTemplateTreeController` removed.
```

**Review by eshanrnh — COMMENTED** [review]
Looks good, @AndyButland. I've added couple of suggestions.

**Comment by AndyButland** [comment]
Thanks Esha.  They all look good to me so have added them to the PR.

**Comment by sofietoft** [comment]
Hi @AndyButland !
Just added a single suggestion to shorten a sentence a little bit ☝️ 

Regarding the Deploy/Extending article - should that be versioned for Umbraco 11, or do the changes also apply to the older versions? 🤔

**Comment by AndyButland** [comment]
Hi @sofietoft - thanks (I can't see the suggestion though?).

I have updated the "Extending" article for V11 yes.  It was a big file though, with a small update, so I didn't create a new versioned file and instead just add the note about the change for V11.  Really just not to have what would then be three files, mostly the same, to keep up to date.  But we can do that if you prefer and just accept we need to do that... so please let me know.

**Review by sofietoft — COMMENTED** [review]
Ah! The suggestion was still pending. My bad! Here you go 😅

**Comment by sofietoft** [comment]
I didn't notice the note you made about the changes in Umbraco 11 - I think with that, there's not need to create a new version of the article 💪 
You're right that it does seem like a lot to have to do for those few lines!

I'll make sure this is merged in!

**Pull request merged**